server/etcdserver/api/v3rpc/grpc.go

```diff
@@ -39,18 +40,29 @@ func Server(s *etcdserver.EtcdServer, tls *tls.Config, interceptor grpc.UnarySer
 	if tls != nil {
 		opts = append(opts, grpc.Creds(credentials.NewTransportCredential(tls)))
 	}
+
+	serverMetrics := grpc_prometheus.NewServerMetrics()
```

Comment:
Probably we don't need `constructExtensiveMetricsInterceptors` any more?
```suggestion
	var opts []grpc_prometheus.ServerMetricsOption
	if s.Cfg.Metrics == "extensive" {
		opts = append(opts, grpc_prometheus.WithServerHandlingTimeHistogram())
	}
	serverMetrics := grpc_prometheus.NewServerMetrics(opts)
```

Reply:
I can't seem to get this to work @ahrtr maybe we can do a follow up?

Reply:
It (the suggested change) should work, please see https://github.com/etcd-io/etcd/pull/19195#issuecomment-2624840140

Reply:
Removed the integration test case TestAllMetricsGenerated as mentioned in https://github.com/etcd-io/etcd/pull/19195#issuecomment-2627486682

cc @dims

Reply:
@ahrtr sounds good. thanks!